src/users/containers/edit-user-permissions/edit-user-permission.component.ts

```diff
@@ -0,0 +1,151 @@
+    isPuiFinanceManager: boolean;
+    userId: string;
+
+    userSubscription: Subscription;
+    dependanciesSubscription: Subscription;
+    editPermissionSuccessSubscription: Subscription;
+    editPermissionServerErrorSubscription: Subscription;
+    backUrl: string;
+
+    summaryErrors: {isFromValid: boolean; items: { id: string; message: any; }[]; header: string};
+    permissionErrors: {isInvalid: boolean; messages: string[] };
+
+    constructor(
+      private userStore: Store<fromStore.UserState>,
+      private routerStore: Store<fromRoot.State>,
+      private actions$: Actions
+    ) { }
+
+    ngOnInit(): void {
+
+      this.editPermissionSuccessSubscription = this.actions$.pipe(ofType(fromStore.EDIT_USER_SUCCESS)).subscribe(() => {
+        this.routerStore.dispatch(new fromRoot.Go({ path: [`users/user/${this.userId}`] }));
+      });
+
+      this.editPermissionServerErrorSubscription = this.actions$.pipe(ofType(fromStore.EDIT_USER_SERVER_ERROR)).subscribe(() => {
+        this.routerStore.dispatch(new fromRoot.Go({ path: [`service-down`] }));
+      });
+
+      this.isLoading$ = this.userStore.pipe(select(fromStore.getGetUserLoading));
+
+      this.dependanciesSubscription = combineLatest([
+      this.routerStore.pipe(select(fromRoot.getRouterState)),
+      this.userStore.pipe(select(fromStore.getGetUserLoaded))
+    ]).subscribe(([route, users]) => {
+      if (!users) {
+        this.userStore.dispatch(new fromStore.LoadUsers());
+      }
+      this.userId = route.state.params.userId;
+      this.user$ = this.userStore.pipe(select(fromStore.getGetSingleUser, { userIdentifier: this.userId }));
+      this.backUrl = this.getBackurl(this.userId);
+    });
+
+      this.userSubscription = this.user$.subscribe((user) => {
+        this.user = user;
+        this.isPuiCaseManager = this.getIsPuiCaseManager(user);
+        this.isPuiOrganisationManager = this.getIsPuiOrganisationManager(user);
+        this.isPuiUserManager = this.getIsPuiUserManager(user);
+        this.isPuiFinanceManager = this.getIsPuiFinanceManager(user);
+
+        this.editUserForm = this.getFormGroup(this.isPuiCaseManager,
+          this.isPuiUserManager, this.isPuiOrganisationManager, this.isPuiFinanceManager,
+          checkboxesBeCheckedValidator);
+      });
+
+    }
+
+  getBackurl(userId: string) {
+    return `/users/user/${userId}`;
+  }
+
+  getFormGroup(isPuiCaseManager, isPuiUserManager, isPuiOrganisationManager, isPuiFinanceManager, validator: any): FormGroup {
+    return new FormGroup({
+      roles: new FormGroup({
+        'pui-case-manager': new FormControl(isPuiCaseManager),
+        'pui-user-manager': new FormControl(isPuiUserManager),
+        'pui-organisation-manager': new FormControl(isPuiOrganisationManager),
+        'pui-finance-manager': new FormControl(isPuiFinanceManager)
+      }, checkboxesBeCheckedValidator())
+    });
+  }
+
+  getIsPuiCaseManager(user: any): boolean {
+    return user && user.manageCases === 'Yes';
```

Comment:
this logic and all below should live within the selectors.. we should have business logic in the componet in the ngrx architecture